exercises/allergies/examples/src/Allergies.purs

```diff
@@ -0,0 +1,31 @@
+module Allergies
+  ( allergicTo
+  , list
+  ) where
+
+import Prelude
+import Data.Array (filter)
+import Data.Int.Bits ((.&.))
+import Data.Map (fromFoldable, lookup)
+import Data.Maybe (fromMaybe)
+import Data.Tuple (Tuple(Tuple), fst)
+
+weights :: Array (Tuple String Int)
+weights = 
+  [ Tuple "eggs"         1
+  , Tuple "peanuts"      2
+  , Tuple "shellfish"    4
+  , Tuple "strawberries" 8
+  , Tuple "tomatoes"     16
+  , Tuple "chocolate"    32
+  , Tuple "pollen"       64
+  , Tuple "cats"         128
+  ]
+
+allergicTo :: Int -> String -> Boolean
+allergicTo s i = fromMaybe false $ do
+  w <- lookup i (fromFoldable weights)
+  pure $ s .&. w == w
```

Comment:
It might be, I was stating my personal preference :P 
I also prefer qualified function names, which I think also isn't popular in the Purescript community.